### DISCUSSION TITLE
Why is HMR being called on production?

### DISCUSSION THREAD

**Discussion by moshie** [vercel/next.js#39464] [Help] [answered]
I don't know why but since updating NextJS to latest (`12.2.4`) it's decided to start calling HMR on our production environment?

<img width="558" alt="Screenshot 2022-08-10 at 15 46 07" src="https://user-images.githubusercontent.com/3974301/183932949-e4944d59-f828-4780-9085-d61040ce3661.png">

I am convinced there is a memory leak in NextJS somewhere. I also noticed crazy increases in memory usage causing our app to crash?

<img width="637" alt="Screenshot 2022-08-10 at 15 51 56" src="https://user-images.githubusercontent.com/3974301/183934905-f92b3f52-f68c-4fca-b632-d2fd1de19cbb.png">

**Comment by ijjk** [comment, marked as answer]
@moshie if `/_next/webpack-hmr` is being called it sounds like a non-production build was deployed that would also explain increased memory usage as dev inherently uses more memory than production.

**Reply by moshie** [reply]
I pulled down the image and can see that the build has production environment variables in it and the NODE_ENV is set to production?

**Reply by moshie** [reply]
@ijjk Little confused why it would start spamming calls to `/_next/webpack-hmr` when all we did was update NextJS?

```
- "next": "~12.2.2",
+ "next": "~12.2.3",
```

**Reply by ijjk** [reply]
We haven't seen this on any other projects so a minimal reproduction would be needed to help track down what could be causing this for your project.

**Comment by cjmp99** [comment]
Something is wrong with this because in my case the request never ends and it causes my application to crash!!!

**Reply by cjmp99** [reply]
![Captura](https://user-images.githubusercontent.com/83384741/233393416-9fd43473-f11c-40bb-bde7-9e679e38c6fc.PNG)

**Comment by moonthug** [comment]
For whoever this may help... I was calling `next` instead of `next start` on production (I think I followed from a random tutorial). This seems to run NextJS in `dev` mode and overrides any `NODE_ENV` set.

It would be great if it was documented [here](https://nextjs.org/docs/app/api-reference/next-cli) or at least gave some kind of messaging when running in dev mode (I would have expected a cli to fail and show usage without a given command)
